File: public/public-persistence-jooq/src/main/java/ch/difty/scipamato/publ/persistence/keyword/JooqKeywordService.java
package ch.difty.scipamato.publ.persistence.keyword;

import java.util.List;

import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Service;

import ch.difty.scipamato.publ.entity.Keyword;
import ch.difty.scipamato.publ.persistence.api.KeywordService;

/**
 * jOOQ specific implementation of the {@link KeywordService} interface.
 *
 * @author u.joss
 */
@Service
public class JooqKeywordService implements KeywordService {

    private final KeywordRepository repo;

    public JooqKeywordService(@NotNull final KeywordRepository repo) {
        this.repo = repo;
    }

    @NotNull
    @Override
    public List<Keyword> findKeywords(@NotNull final String languageCode) {
        return repo.findKeywords(languageCode);
    }
}

File: public/public-persistence-jooq/src/main/kotlin/ch/difty/scipamato/publ/persistence/keyword/JooqKeywordService.kt
package ch.difty.scipamato.publ.persistence.keyword

import ch.difty.scipamato.publ.entity.Keyword
import ch.difty.scipamato.publ.persistence.api.KeywordService
import org.springframework.stereotype.Service

/**
 * jOOQ specific implementation of the [KeywordService] interface.
 */
@Service
class JooqKeywordService(private val repo: KeywordRepository) : KeywordService {
    override fun findKeywords(languageCode: String): List<Keyword> = repo.findKeywords(languageCode)
}
